march_gazebo_plugins/src/com_controller.cpp

```diff
@@ -0,0 +1,210 @@
+#include <gazebo/common/common.hh>
+#include <gazebo/physics/physics.hh>
+#include <march_shared_resources/GaitActionGoal.h>
+#include <ros/callback_queue.h>
+#include <ros/ros.h>
+
+namespace gazebo
+{
+class ComController : public ModelPlugin
```

Comment:
You could also implement the [`Reset`](http://osrf-distributions.s3.amazonaws.com/gazebo/api/dev/classgazebo_1_1ModelPlugin.html#a4c4ba0ffe635d14b93794268bd8e5995) method to be able to reset the plugin and the [`Init`](http://osrf-distributions.s3.amazonaws.com/gazebo/api/dev/classgazebo_1_1ModelPlugin.html#a3e0b39e1326de703012f81ac2be7feba) method to initialize variables.

Reply:
It shouldn't be necessary to reset the plugin. Init does not really add much to load, I could move some initializations there, but might as well leave them in Load